modules/thermo-cycler/thermo-cycler-arduino/lid.cpp

```diff
@@ -151,29 +197,42 @@ void Lid::move_millimeters(float mm, bool top_switch, bool bottom_switch)
   {
     _motor_step(dir);
     _update_acceleration();
-    if (top_switch && _is_open_switch_pressed()) return;
-    if (bottom_switch && _is_closed_switch_pressed()) return;
+    if (dir)
+    {
+      if (_is_cover_switch_pressed)
+      {
+        return true;
+      }
+    }
+    else
+    {
+      if(_is_bottom_switch_pressed)
+      {
+        return true;
+      }
+    }
   }
   motor_off();
+  return false;
 }
 
 void Lid::open_cover()
 {
-  if (_is_open_switch_pressed())
+  if (_is_cover_switch_pressed)
   {
      return;
   }
-  move_millimeters(300, true, false);
+  move_millimeters(300);
```

Comment:
This and the `close_cover` distance as magic numbers might want to be in headers or something I guess. Not a big thing they'll still be magic wherever they are